src/React/Resources/assets/src/render_controller.ts

```diff
@@ -0,0 +1,78 @@
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+'use strict';
+
+import React, { ReactElement } from 'react';
+import { Controller } from '@hotwired/stimulus';
+
+export default class extends Controller {
+    readonly componentValue: string;
+    readonly propsValue: object;
+
+    static values = {
+        component: String,
+        props: Object,
+    };
+
+    connect() {
+        this._dispatchEvent('react:connect', { component: this.componentValue, props: this.propsValue });
+
+        // Use a timeout to avoid mounting and demounting right after
+        if ((this.element as any).timeout) {
+            clearTimeout((this.element as any).timeout);
+        }
+
+        (this.element as any).timeout = setTimeout(() => {
+            const component = window.resolveReactComponent(this.componentValue);
+            this._renderReactElement(React.createElement(component, this.propsValue, null));
+
+            this._dispatchEvent('react:mount', {
+                componentName: this.componentValue,
+                component: component,
+                props: this.propsValue,
+            });
+        }, 50);
+    }
+
+    disconnect() {
+        (this.element as any).unmount();
+        this._dispatchEvent('react:unmount', { component: this.componentValue, props: this.propsValue });
+    }
+
+    _renderReactElement(reactElement: ReactElement) {
+        if (parseInt(React.version) >= 18) {
+            // React 18+ rendering
+            // eslint-disable-next-line @typescript-eslint/no-var-requires
+            const root = require('react-dom/client').createRoot(this.element);
+            root.render(reactElement);
+
+            // Register a way to unmount this element
+            (this.element as any).unmount = () => {
+                root.unmount();
+            };
+
+            return;
+        }
+
+        // Up to React 17 rendering
```

Comment:
Should we really support old versions of React in the new code we introduce? Thr React team only supports the current major version: https://github.com/reactjs/reactjs.org/issues/1745#issuecomment-466767389

 I'm not against supporting old versions, but I feel that it isn't worth it.

Reply:
That may be true indeed